### PR TITLE
Preinstall openssl 1.1.1d

### PR DESCRIPTION
**Pull request by ryw**
openssl CVEs were [reported by Truata](https://github.com/astronomer/issues/issues/1172) in the following containers:

* c101-astronomer-astro-ui.astronomer-ns
* c101-astronomer-cli-install.astronomer-ns
* c101-astronomer-commander.astronomer-ns
* c101-astronomer-elasticsearch-nginx.astronomer-ns
* c101-astronomer-houston.astronomer-ns
* c101-astronomer-nginx-default-backend.astronomer-ns

[CVEs were fixed in 1.1.1d](https://git.alpinelinux.org/aports/commit/?h=v3.11.5&id=09a199deeac384bd1f22bb26c2ec5b3bd60257a2)

If we preinstall `OpenSSL 1.1.1d` these downstream images should be good, right?

### CODE REVIEW

File: base/Dockerfile [hunk 1]
```diff
@@ -32,3 +32,8 @@ RUN addgroup -S ${ASTRONOMER_GROUP} \
 RUN apk update \
 	&& apk add --no-cache ca-certificates \
 	&& update-ca-certificates
+
+# Resolves CVE-2019-1549 + CVE-2019-1563 (OpenSSL 1.1.1d)
+RUN \
+	apk add --update openssl && \
+	rm -rf /var/cache/apk/*
```
Comment (schnie) [inline base/Dockerfile:39]: @andriisoldatenko should we combine this with the previous statement or use 2 layers? 

@ryw @andriisoldatenko we should test that this doesn't break any of our downstream images right?

**Comment by andriisoldatenko** [comment]
it will work for:

> c101-astronomer-astro-ui.astronomer-ns
> c101-astronomer-cli-install.astronomer-ns
> c101-astronomer-commander.astronomer-ns
> c101-astronomer-houston.astronomer-ns
> 
but doesn't work for:
> c101-astronomer-elasticsearch-nginx.astronomer-ns
> c101-astronomer-nginx-default-backend.astronomer-ns

https://github.com/astronomer/ap-vendor/blob/77130434cba17cd147d647e027b09b40259b878e/nginx-es/Dockerfile#L16

File: base/Dockerfile (hunk 1 above)
Comment (andriisoldatenko) [reply base/Dockerfile:39]: >should we combine this with the previous statement or use 2 layers?
yes we should

**Comment by andriisoldatenko** [comment]
also we have to bump version.txt

File: base/Dockerfile (hunk 1 above)
Comment (ryw) [reply base/Dockerfile]: fixed

**Comment by ryw** [comment]
> also we have to bump version.txt

base/version.txt says 3.11.5 - what am i missing?

**Comment by ryw** [comment]
> it will work for:
> 
> > c101-astronomer-astro-ui.astronomer-ns
> > c101-astronomer-cli-install.astronomer-ns
> > c101-astronomer-commander.astronomer-ns
> > c101-astronomer-houston.astronomer-ns
> 
> but doesn't work for:
> 
> > c101-astronomer-elasticsearch-nginx.astronomer-ns
> > c101-astronomer-nginx-default-backend.astronomer-ns
> 
> https://github.com/astronomer/ap-vendor/blob/77130434cba17cd147d647e027b09b40259b878e/nginx-es/Dockerfile#L16

why don't we use our base image as starting point for these images?

**Comment by schnie** [comment]
If there are any vendor images using alpine, I'd think we could use our base image.

**Comment by ryw** [comment]
opened https://github.com/astronomer/ap-vendor/pull/23 too

**Comment by andriisoldatenko** [comment]
LGTM

**Comment by ryw** [comment]
@schnie did i name the version correctly on this?

**Comment by ryw** [comment]
Actually i now realize that 1.1.1d is already installed in 3.11.5 - so closing this as not necessary

**Comment by andriisoldatenko** [comment]
@ashb PTAL

**Comment by andriisoldatenko** [comment]
from @ashb :
>ash  3:34 PM
@andrii https://www.openssl.org/news/secadv/20200421.txt -- we'll need to bump OpenSSL in all our images.





3:34
(Alpine 3.10 thankfully does package Openssl 1.1.1g https://pkgs.alpinelinux.org/package/v3.10/main/armhf/openssl)

```
/ # apk add --update openssl
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
(1/1) Installing openssl (1.1.1g-r0)
Executing busybox-1.31.1-r9.trigger
OK: 6 MiB in 15 packages
/ # openssl
OpenSSL> / # ^C
/ # openssl version
OpenSSL 1.1.1g  21 Apr 2020 (Library: OpenSSL 1.1.1d  10 Sep 2019)
```

**Comment by ryw** [comment]
ha that's pretty funny :)
is `3.11.5-2` the correct way to name this version?

**Comment by andriisoldatenko** [comment]
@ryw yeah i like it :)

**Comment by ryw** [comment]
we should add to README a list of all of our images that use ap-base

**Comment by ryw** [comment]
added list to README

**Pull request merged**